flashinfer/trtllm_flavored_gemm.py

```diff
@@ -0,0 +1,214 @@
+    metainfo = get_cubin(
+        f"{include_path}/{header_name}.h",
+        MetaInfoHash.TRTLLM_GEN_GEMM,
+    )
+    # make sure "flashinferMetaInfo.h" is downloaded or cached
+    assert metainfo, f"{header_name}.h not found"
+    return gen_jit_spec(
+        "trtllm_gemm",
+        [
+            jit_env.FLASHINFER_CSRC_DIR / "trtllm_flavored_gemm_runner.cu",
+        ],
+        extra_cuda_cflags=[
+            "-DTLLM_GEN_EXPORT_INTERFACE",
+            "-DTLLM_ENABLE_CUDA",
+            f'-DTLLM_GEN_GEMM_CUBIN_PATH=\\"{ArtifactPath.TRTLLM_GEN_GEMM}\\"',
+        ]
+        + sm100a_nvcc_flags,
+        # link "include" sub-directory in cache
+        extra_include_paths=[jit_env.FLASHINFER_CUBIN_DIR / include_path],
+        extra_ldflags=["-lcuda"],
+    )
+
+
+@functools.cache
+def get_trtllm_flavored_gemm_module():
+    mod = gen_trtllm_gen_flavored_gemm_module()
+    op = mod.build_and_load()
+    setup_cubin_loader(str(mod.get_library_path()))
+
+    class TrtllmFlavoredGemmRunner(TunableRunner):
+        def get_valid_tactics(
+            self,
+            inputs: List[torch.Tensor],
+            profile: OptimizationProfile,
+        ) -> List[int]:
+            a_tensor_index = 0
+            b_tensor_index = 1
+
+            # NOTE : expects  A=MxK, B=(K//B)xNxB, out=MxN
+            a = profile.get_opt_shapes()[a_tensor_index]
+            b = profile.get_opt_shapes()[b_tensor_index]
+            m = a[0]
+            n = b[1]
+            k = a[1]
+            (
+                a,
+                b,
+                global_scale,
+                out,
+            ) = inputs
+            type_e4m3 = 1
+            type_bf16 = 2
+            valid_tactics = list(
+                op.trtllm_flavored_gemm_tactics(m, n, k, type_e4m3, type_bf16)
+            )
+            return valid_tactics
+
+        def forward(
+            self,
+            inputs: List[torch.Tensor],
+            tactic: int = -1,
+            do_preparation: bool = False,
+            **kwargs,
+        ) -> torch.Tensor:
+            (
+                a,
+                b,
+                global_scale,
+                out,
+            ) = inputs
+            if tactic < 0:
+                return out
+            m = a.shape[0]
+            n = b.shape[1]
+            k = a.shape[1]
+            workspace_size = op.get_workspace_size_in_bytes(m, n, k, tactic)
+            workspace_buffer = _get_cache_buf(
+                "trllm_flavored_gemm", workspace_size, a.device
+            )
+            op.trtllm_flavored_gemm(
+                workspace_buffer,
+                a,
+                b,
+                global_scale,
+                out,
+                tactic,
+            )
+            return out
+
+    def gemm_runner():
+        return TrtllmFlavoredGemmRunner()
+
+    # Register the module
+    return SimpleNamespace(
+        gemm_runner=gemm_runner,
+    )
+
+
+def trtllm_flavored_gemm(
```

Comment:
Please align APIs between FP8 gemms and FP4 gemms: https://github.com/flashinfer-ai/flashinfer/blob/main/flashinfer/gemm.py#L2012

I don't mind if we follow this style for both FP8 and FP4, or we follow the FP4 gemm's style for both FP8 and FP4, but at least FP8 and FP4 should have similar APIs to avoid further confusions